src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -214,19 +235,37 @@ public void DeleteSearchParameter(string url, bool calculateHash = true)
 
         public async Task Handle(SearchParametersUpdatedNotification notification, CancellationToken cancellationToken)
         {
-            _logger.LogInformation("SearchParameterDefinitionManager: Search parameters updated");
-            CalculateSearchParameterHash();
-            await _mediator.Publish(new RebuildCapabilityStatement(RebuildPart.SearchParameter), cancellationToken);
+            try
+            {
+                _logger.LogInformation("SearchParameterDefinitionManager: Search parameters updated");
+                CalculateSearchParameterHash();
+                await _mediator.Publish(new RebuildCapabilityStatement(RebuildPart.SearchParameter), cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error calculating search parameter hash");
+                await _mediator.Publish(new ImproperBehaviorNotification("Error calculating search parameter hash"), cancellationToken);
+            }
         }
 
         public async Task Handle(StorageInitializedNotification notification, CancellationToken cancellationToken)
         {
-            _logger.LogInformation("SearchParameterDefinitionManager: Storage initialized");
-            await EnsureInitializedAsync(cancellationToken);
+            try
+            {
+                _logger.LogInformation("SearchParameterDefinitionManager: Storage initialized");
+                await EnsureInitializedAsync(cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Error initializing search parameters");
+                await _mediator.Publish(new ImproperBehaviorNotification("Error initializing search parameters"), cancellationToken);
+            }
         }
 
-        private async Task LoadSearchParamsFromDataStore(CancellationToken cancellationToken)
+        private Task LoadSearchParamsFromDataStore(CancellationToken cancellationToken)
         {
+            throw new NotImplementedException("LoadSearchParamsFromDataStore is not implemented");
+            /*
```

Comment:
Should we keep this part of the code commented?

Reply:
No, that was for testing. I'll revert that.